cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -182,6 +184,54 @@ public void recordProgramStop(Id.Program program, String pid, long stopTs, Progr
     }
   }
 
+  public RunRecord getRun(Id.Program program, final String runid) {
```

Comment:
This method is called from handler. runid is gotten from path parameter, so it cannot be null. If it is null then user will get 404.